### ISSUE TITLE
Rule Exclusions for Wordpress Password-Reset Page not working. Caused by "Simple Cloudflare Turnstile" plugin. Argument value too long. Desperately need help!

### ISSUE THREAD

**Issue by Danrancan** [coreruleset/coreruleset#3249]
### Description

I am running a LEMP server with the latest mainline Nginx, and the latest ngxmodsecurity module. The issue I am having is specific to mobile devices, particularly, Safari on iOS. I am getting a FALSE POSITIVE on my Wordpress "Reset Password" Page, while using the "Simple Cloudflare Turnstile" Wordpress plugin to protect my Login, Registration, and Password reset pages with a Cloudflare turnstile captcha. On iOS Safari, when I navigate to https://www.mcmo.is/my-account/lost-password/ and enter my Woo Commerce user email address to reset my password, I instantly get redirected to a "403 Forbidden" error page, and am not able to get a success page from submitting my email properly. Therefore, Modsecurity is breaking the password reset page by preventing me from submitting my email address to the password recovery tool while using the "Simple Cloudflare Turnstile" plugin.

For the life of me, I cannot figure out the proper rule exclusions to prevent this page from giving a 403 upon email submission. I desperately need some help and would be very grateful to get some answers out there. Thanks so much in advance for your help and efforts in helping me find the proper rules exclusions.

### How to reproduce the misbehavior

Go to https://www.mcmo.is/my-account/lost-password/  in your iOS Safari web browser, and request a password reset by submitting your username or email address in the password reset box. Result: a 403 error page will appear.

### Current Rule Exclusions (Not Working)

```
# iOS Safari: This rule attempts to prevent a 403 error white page after requesting a password reset
SecRule REQUEST_URI "@streq /my-account/lost-password" \
    "id:1064,\
    phase:1,\
    pass,\
    nolog,\
    ctl:ruleRemoveById=920370,\
    ctl:ruleRemoveById=949110"

# iOS Safari: This rule attempts to prevent a 403 error white page after requesting a password reset
SecRule REQUEST_FILENAME "@streq /my-account/lost-password" \
    "id:1065,\
    phase:1,\
    pass,\
    nolog,\
    ctl:ruleRemoveById=920370,\
    ctl:ruleRemoveById=949110"
```

### Logs

MODSEC AUDIT LOG:
```
---iso7bGYz---H--
ModSecurity: Warning. Matched "Operator `Gt' with parameter `400' against variable `ARGS:cf-turnstile-response' (Value: `0._Sqywz3fBpJY2g4BOtI_0NC0IwNj-FFIISwXRcYP3BOh6CCyp3Uy3wMkz_Ko-x3LyMnYXirxSjZ1_dzs70HpwnEvHrlL0YYx5b (396 characters omitted)' ) [file "/etc/nginx/modsec/coreruleset-3.3.4/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf"] [line "804"] [id "920370"] [rev ""] [msg "Argument value too long"] [data "ARGS:cf-turnstile-response=496"] [severity "2"] [ver "OWASP_CRS/3.3.4"] [maturity "0"] [accuracy "0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-protocol"] [tag "paranoia-level/1"] [tag "OWASP_CRS"] [tag "capec/1000/210/272"] [hostname "10.10.10.2"] [uri "/my-account/lost-password/"] [unique_id "168893318715.711653"] [ref "v761,496t:length"]
ModSecurity: Access denied with code 403 (phase 2). Matched "Operator `Ge' with parameter `5' against variable `TX:ANOMALY_SCORE' (Value: `5' ) [file "/etc/nginx/modsec/coreruleset-3.3.4/rules/REQUEST-949-BLOCKING-EVALUATION.conf"] [line "81"] [id "949110"] [rev ""] [msg "Inbound Anomaly Score Exceeded (Total Score: 5)"] [data ""] [severity "2"] [ver "OWASP_CRS/3.3.4"] [maturity "0"] [accuracy "0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-generic"] [hostname "10.10.10.2"] [uri "/my-account/lost-password/"] [unique_id "168893318715.711653"] [ref ""]

---iso7bGYz---J--

---iso7bGYz---K--

---iso7bGYz---Z--

---HNqPuAN9---A--
[09/Jul/2023:15:07:39 -0500] 168893325954.979268 66.63.167.204 50652 10.10.10.2 443
---HNqPuAN9---B--
POST /my-account/lost-password/ HTTP/2.0
user-agent: Mozilla/5.0 (iPhone; CPU iPhone OS 16_5_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.5.1 Mobile/15E148 Safari/604.1
sec-fetch-site: same-origin
referer: https://www.mcmo.is/my-account/lost-password/
origin: https://www.mcmo.is
accept-encoding: gzip, deflate, br
cookie: _pk_id.1.37a4=4575d18aec64a3e5.1688933035.; _pk_ses.1.37a4=1; image_random_value_1_1=dc4eae381f13f576feb4455ea653e1ea
content-length: 667
accept-language: en-US,en;q=0.9
accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
content-type: application/x-www-form-urlencoded
sec-fetch-mode: navigate
host: www.mcmo.is
sec-fetch-dest: document

---HNqPuAN9---C--
user_login=randa.dan%40gmail.com&cf-turnstile-response=0.TICkTvf6m7Oow54o7CgE-MOF9k1x0kpFA8Z7FIiu0aAmvkOyuSMNN9aa6tl7bxzrTfNyt_Mv4y3rK2Cj7El2QJ8BlXWXanV9qL-NJGQ_jS2l51UGUu-9PCGdDjNehkVIj0vtaFs55CMX3uJr4Ja10ZSDOgBSZv4Af8VnMSak-2kFJ5iFYgsupdc-5nLJNjDBspRRWvxjRQqc0IFxsw6jnQG6002-qUfxwFK8bZBKZ0TWp9YynnC7MO8F4M_SVEUTr-3xzhP3h__iS8WdxSzjQ3LwRz2f5bwbwwzQIpHz5QTWTblCXJFd3Ykl52zf8AFsIR5HVQHMugeuurnWLUpGl2cZ1JOl3CZkE7q4p-eRMUe4cMwutpfZ2jrCX373CeEjGPSYmNlGuDBMeQA8mK4E9g.bfZ2QHpwT8LehC6SCapZTQ.bda440e104415f25caa2d13067ec3ed2a77422ca85042b402092e32a126ec0ad&wc_reset_password=true&woocommerce-lost-password-nonce=819a9ee343&_wp_http_referer=%2Fmy-account%2Flost-password%2F

---HNqPuAN9---E--
\xa1\x88\x04\x00 :\xb7\xceF\xe8\x84\x06\x0c\xf2\xa3)X\xc4\x82\x1bI=Y\xc8\x99]2\x92L\x0a\x0aZ\xa37|\xdc\xbe5I\xe4bPIXo\xd5\x05mi!\xeb\xcdn\xd3!\x14&\xcb$\x98d!\xd8Q\x19\xc5\x95\xca\xc5\xaar\x8c\x1bY\xd6\x80\xf0\xfa\xdc\xfe\xb8kD\xd3l\x00

---HNqPuAN9---F--
HTTP/2.0 403
Server: nginx
Date: Sun, 09 Jul 2023 20:07:39 GMT
Content-Type: text/html
X-Content-Type-Options: nosniff
Connection: close
X-XSS-Protection: 1; mode=block
Strict-Transport-Security: max-age=63072000; includeSubDomains; preload
Content-Encoding: br
Content-Security-Policy: default-src * data: 'unsafe-eval' 'unsafe-inline'
Referrer-Policy: no-referrer-when-downgrade
x-frame-options: SAMEORIGIN

---HNqPuAN9---H--
ModSecurity: Warning. Matched "Operator `Gt' with parameter `400' against variable `ARGS:cf-turnstile-response' (Value: `0.TICkTvf6m7Oow54o7CgE-MOF9k1x0kpFA8Z7FIiu0aAmvkOyuSMNN9aa6tl7bxzrTfNyt_Mv4y3rK2Cj7El2QJ8BlXWXanV9qL (396 characters omitted)' ) [file "/etc/nginx/modsec/coreruleset-3.3.4/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf"] [line "804"] [id "920370"] [rev ""] [msg "Argument value too long"] [data "ARGS:cf-turnstile-response=496"] [severity "2"] [ver "OWASP_CRS/3.3.4"] [maturity "0"] [accuracy "0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-protocol"] [tag "paranoia-level/1"] [tag "OWASP_CRS"] [tag "capec/1000/210/272"] [hostname "10.10.10.2"] [uri "/my-account/lost-password/"] [unique_id "168893325954.979268"] [ref "v761,496t:length"]
ModSecurity: Access denied with code 403 (phase 2). Matched "Operator `Ge' with parameter `5' against variable `TX:ANOMALY_SCORE' (Value: `5' ) [file "/etc/nginx/modsec/coreruleset-3.3.4/rules/REQUEST-949-BLOCKING-EVALUATION.conf"] [line "81"] [id "949110"] [rev ""] [msg "Inbound Anomaly Score Exceeded (Total Score: 5)"] [data ""] [severity "2"] [ver "OWASP_CRS/3.3.4"] [maturity "0"] [accuracy "0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-generic"] [hostname "10.10.10.2"] [uri "/my-account/lost-password/"] [unique_id "168893325954.979268"] [ref ""]

---HNqPuAN9---J--

---HNqPuAN9---K--

---HNqPuAN9---Z--
```

### Your Environment

* CRS version: v3.3.4
* Paranoia level setting: PL2
* ModSecurity_Nginx: v1.0.3
* libmodsecurity3_for_Nginx_v1.25.1_Mainline
* Web Server and version: Nginx v1.25.1 Mainline
* Operating System and version: Ubuntu Server 20.04 for Raspberry Pi 4

### Confirmation

[X ] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

**Comment by azurit** [comment]
Hi @Danrancan. You can resolve this particular problem by easier way - just find rule ID `900320` in your `crs-setup.conf`, uncomment it and raise the limit - try 500 or so.

**Comment by Danrancan** [comment]
> Hi @Danrancan. You can resolve this particular problem by easier way - just find rule ID `900320` in your `crs-setup.conf`, uncomment it and raise the limit - try 500 or so.

WOW! I didn't even realize that I had originally set the value of the 900320 setting to a 400 argument limit in crs-setup.conf. By default that setting was off, but I turned it on to increase security. After that, I forgot that I turned it on and didn't realize it was relative to these rules until you pointed it out just now.

Essentially,
Problem solved! Thank you so so much for your response and help! I was stressing out over this. Big thanks!
